### PR TITLE
Add doc for explaining Finder Content Item

### PR DESCRIPTION
**Pull request by tommyp**
First pass at a doc for what the Finder Content Item should contain. Leaving the facets for now until https://github.com/alphagov/finder-frontend/pull/148 is merged.

### CODE REVIEW

File: docs/finder-content-item.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# The Finder Content Item Format
+
+A Finder Content Item is a specification of the [Content Item](https://github.com/alphagov/content-store/blob/master/doc/content_item_fields.md). This guide explains what goes in the details hash of the ContentItem and why.
+
+The Finder Content Item is used by Finder Frontend to render the Finder page. Some of the details Most of what it uses goes in the `details` hash.
```
Comment (evilstreak) [inline docs/finder-content-item.md]: There are some spare words here. "Some of the details Most of what"

File: docs/finder-content-item.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+# The `details` hash
+
+## `beta`
+
+A boolean. Required.
```
Comment (evilstreak) [inline docs/finder-content-item.md]: Why is this required? Can't we just default it to false?

Comment (tommyp) [inline docs/finder-content-item.md]: We could, but I thought I'd pass this on what currently is required and change it when I refactor the App.

File: docs/finder-content-item.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+A flag used to decide if the Beta banner should be rendered.
+
+## `beta_message`
+
+A string. Required.
```
Comment (evilstreak) [inline docs/finder-content-item.md]: If `beta` is false is this still required? Even if `beta` is true, isn't there a default beta message that can be used if this isn't provided?

File: docs/finder-content-item.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+Can contain HTML. If `beta` is true, `beta_message` will be passed to the beta banner.
+
+## `document_noun`
+
+A string. Required.
+
+The lowercase singular version of whatever format the Finder is using. For example: [`/cma-cases`](https://www.gov.uk/cma-cases) has a `document_noun` of `case`, [`/aaib-report`](https://www.gov.uk/aaib-reports) has a `document_noun` of `report`.
```
Comment (evilstreak) [inline docs/finder-content-item.md]: It would be good to describe the ways this string is used. Is it only used for constructing the filter sentence?

File: docs/finder-content-item.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+## `document_type`
+
+A string. Required.
+
+[snake_case](http://en.wikipedia.org/wiki/Snake_case) string which tells Finder Frontend what doctype to limit the search to in Rummager. It must match the name of the file describing the doctype [in Rummager](https://github.com/alphagov/rummager/tree/master/config/schema/default/doctypes).
```
Comment (evilstreak) [inline docs/finder-content-item.md]: Given that it needs to match the doctype in Rummager, we probably don't need to stipulate here that should be snake_case.

File: docs/finder-content-item.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+Used to decide if the link to the email alert signup page should be displayed
+
+## `format_name`
+
+A string. Optional.
+
+Not specifically used by the Finder, but used by [Specialist Frontend](https://github.com/alphagov/specialist-frontend) to link back to the Finder.
+Usually a singularised version of the title of the Finder - `"Competition and Markets Authority case"` for [`/cma-cases`](https://www.gov.uk/cma-cases) for example.
+However there are edge cases where it's not the same such as `"Medical safety alert"` for [Alerts and recalls for drugs and medical devices].
```
Comment (evilstreak) [inline docs/finder-content-item.md]: Missing a link URL here

File: docs/finder-content-item.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+## `signup_link`
+
+A string. Optional.
+
+If `email_signup_enabled` is set to true, the link being displayed will point to `base_path/email-signup` where `base_path` is from the Finder object. `signup_link` allows you to point it at a different URL, [Drug Safety Update](https://www.gov.uk/drug-safety-update) and [Drug Device Alerts](https://www.gov.uk/drug-device-alerts) are the two which currently use this feature.
+
+## `show_summaries`
+
+A boolean. Required.
```
Comment (evilstreak) [inline docs/finder-content-item.md]: Can't we default this to `false` rather than requiring it, as that's the behaviour for most finders?

Comment (tommyp) [inline docs/finder-content-item.md]: Again, I thought I'd explain what the current case is and alter this when I refactor it.
